### PR TITLE
Setup (WIP)

### PR DESCRIPTION
**Pull request by jamienoss**
Additions:

- [x] dependabot actions workflow
- [x] CI
  - [x] linting
  - [x] pytests
  - [x] docker build & publish (to GitHub packages)
  - [ ] deploy (may punt to follow up PR). **PUNT TO #5**
  - [x] code coverage
  - [x] Status badges
- [x] https://zenodo.org/
    - [x] Need everyone's Orcid IDs. https://orcid.org/
- [x] CITIATION.cff
- [x] Readme
- [x]  fastapi app (example mainly to have something to test)
- [x] pyproject.toml & setup.py
- [x] pip requirements files for both prd and test
- [x] Dockerfile (super simple, mostly example). 
  - [ ] ~Will need to add docs build and build as multistage.~ Repo's now public and using readthedocs.io which handles doc builds.
  - [ ] Add Docker compose instructions to README. **PUNT to #4**
- [x] pre-push githook for running linter.
  - [ ] ~Run pytests also?~ Nah.
- [ ] Code of conduct. **PUNT TO #8**
- [ ] Add regression test framework. **PUNT TO #2**
- [ ] Add smoke test example. **PUNT TO  #3**
- [x] Add Docs framework and skeleton example.
- [ ] Add contents of readme to docs. **PUNT TO  #7.**

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ignore =
+    W605
+
+select =
+    E4
+    E7,
+    E9,
+    W6,
+    F821,
+    F822
```
Comment (jamienoss) [inline .flake8:15]: I need to go over these in another PR. I just grabbed them from what I normally used but haven't reviewed them in a while. See #6

File: README.md [hunk 2]
```diff
@@ -1,2 +1,102 @@
-# base-template
-Base repo template 
+# SSEC-JHU <package_name>
+
+[![CI Status](https://github.com/ssec-jhu/base-template/workflows/CI/badge.svg?branch=main)](https://github.com/ssec-jhu/base-template/actions)
+[![Documentation Status](https://readthedocs.org/projects/base-template/badge/?version=latest)](http://ssec-jhu-base-template.readthedocs.io/en/latest/?badge=latest)
+[![codecov](https://codecov.io/gh/ssec-jhu/base-template/branch/main/graph/badge.svg?token=0KPNKHRC2V)](https://codecov.io/gh/ssec-jhu/base-template)
```
Comment (jamienoss) [inline README.md:5]: These can all be navigated to and are live links. Their status currently reports garbage, i.e., "unknown" as they ref the main branch which'll get updated once this is merged.

Comment (jamienoss) [reply README.md:5]: To view an example of the docs goto https://ssec-jhu-base-template--1.org.readthedocs.build/en/1/

Comment (ehunter68) [reply README.md:5]: :)

**Review by ehunter68 — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+requires = ["setuptools>=61.2", "setuptools_scm[toml]>=7"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["package_name"]  # Replace package_name with actual name (it's just a template).
+zip-safe = false
+include-package-data = true
+
+[tool.setuptools_scm]
+write_to = "package_name/_version.py"  # Replace package_name with actual name (it's just a template).
```
Comment (amitschang) [inline pyproject.toml:40]: add in tox anywhere?

File: .flake8 [hunk 4]
```diff
@@ -0,0 +1,24 @@
+[flake8]
+max-line-length=127
```
Comment (amitschang) [inline .flake8:2]: would vote 120 (simply because idies uses such limit elsewhere), but not a strong vote :smile:

Comment (jamienoss) [reply .flake8:2]: I'll address this in #10.

File: pyproject.toml (hunk 3 above)
Comment (jamienoss) [reply pyproject.toml:45]: @amitschang tox.ini added.
